lib/url_parser.rb

```diff
@@ -53,6 +55,27 @@ def parse_to_full_user_url
     [full_domain, path].join('/')
   end
 
+  def self.case_sensitive?
+    self::CASE_SENSITIVE
+  end
+
+  # This computation is memoized because it is expensive. This prevents use cases which require using
+  # .try_all in a tight loop. However, if this class is required directly (without requiring any subparsers),
+  # this method will memoize an empty array. It is recommended to simply require librariesio-url-parser.rb directly.
+  # This is the default behavior when installing this gem.
+  def self.all_parsers
```

Comment:
Moved this out of private so I can get a list of all the available parser classes.